src/values.rs

```diff
@@ -632,6 +635,32 @@ impl ToJSON for Color {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct ThinkerbellRuleSource {
+    pub name: String,
+    pub source: String,
+}
+
+impl Parser<ThinkerbellRuleSource> for ThinkerbellRuleSource {
+    fn parse(path: Path, source: &mut JSON) -> Result<Self, ParseError> {
+        let name = try!(path.push("name", |path| String::take(path, source, "name")));
+        let script_source = try!(path.push("source", |path| String::take(path, source, "source")));
+        Ok(ThinkerbellRuleSource { name: name, source: script_source })
+    }
+}
+impl ToJSON for ThinkerbellRuleSource {
+    fn to_json(&self) -> JSON {
+        let mut source = vec![
```

Comment:
Actually, I have added stuff that should now let you simply write

```rust
fn to_json(&self) -> JSON {
  vec![...].to_json()
}
```

Can be done later, though, if you prefer.